src/chain/link.rs

```diff
@@ -158,7 +158,7 @@ where
     pub(crate) fn queue_state(&self, qid: QueueId) -> State<R> {
         let queue = self.queue(qid);
         State {
-            access: queue.access,
+            access: self.state.access,
```

Comment:
Currently, the only usage of this function is to generate the barriers for a queue transfer. 

Before, it would only set the access flags used in the queue that does the acquire side of the transfer, and not other queues in the link. I believe this is the correct code.

Reply:
In this case the code that uses this function should be fixed, not the function.

Reply:
And yes. You're right. Acquire operations should include access for whole link.
The same way release operation does.

Reply:
Then the function basically becomes dead code. hrm.

Reply:
¯\\\_(ツ)\_/¯